### ISSUE TITLE
Enable Link-Time Optimization (LTO)

### ISSUE THREAD

**Issue by zamazan4ik** [llMBQll/OmniLED#2]
Hi!

I noticed that in the `Cargo.toml` file Link-Time Optimization (LTO) for the project is not enabled. I suggest switching it on since it will reduce the binary size (always a good thing to have) and will likely improve the application's performance a little bit. If you want to read more about LTO and its possible modes, I recommend starting from [this](https://doc.rust-lang.org/rustc/codegen-options/index.html#lto) Rustc documentation.

I think you can enable LTO only for the Release builds so as not to sacrifice the developers' experience while working on the project since LTO consumes an additional amount of time to finish the compilation routine. If you think that a regular Release build should not be affected by such a change as well, then I suggest adding an additional `dist` or `release-lto` profile where in addition to regular `release` optimizations LTO will also be added. Such a change simplifies life for maintainers and others interested in the project persons who want to build the most optimized version of the application. However, if we enable it on the Cargo profile level for the Release profile, users, who install the application with `cargo install` will get the LTO-optimized version of the game "automatically". E.g., check `cargo-outdated` Release [profile](https://github.com/kbknapp/cargo-outdated/blob/master/Cargo.toml#L48). You also could be interested in other optimization options like `codegen-units = 1`, etc. If you are still concerned by the build times, you can switch to ThinLTO.

Basically, it can be enabled with the following lines:

```
[profile.release]
lto = true
```

I have made quick tests (Fedora 41, Rust 1.82, the latest version of the project) by adding `lto = true` to the Release profile. The binary size reduction is the following:
* `omni-led`: from 15 Mib to 12 Mib
* `audio`: from 3.6 to 2.4 Mib
* `clock`: from 3.6 to 2.4 Mib
* `media`: from 4.1 Mib to 2.8 Mib
* `weather`: from 7.9 Mib to 6.2 Mib

Thank you.

**Comment by llMBQll** [comment]
Great suggestion, it seems like a good tradeoff, especially if only enabled in release builds. I will tinker with the options and see what I like best. Thanks!

**Comment by llMBQll** [comment]
I did some testing and I managed to reproduce your findings, though gains on Windows are a bit smaller (still not insignificant).
In my testing thin LTO did not deliver, unless I went with a single codegen unit that mostly defeats its time advantage.

In the end I will just go with fat LTO with a single codegen unit as I like seeing the smaller numbers :D.

## Windows 11

| binary   | No LTO   | LTO (fat) | LTO (fat) 1 codegen unit | LTO (thin) | LTO (thin) 1 codegen unit |
|----------|----------|-----------|--------------------------|------------|---------------------------|
| omni-led | 8.86 MB  | 8.28 MB   | 8.02 MB                  | 9.05 MB    | 8.43 MB                   |
| audio    | 2.65 MB  | 2.27 MB   | 2.00 MB                  | 2.75 MB    | 2.37 MB                   |
| clock    | 2.54 MB  | 2.17 MB   | 1.93 MB                  | 2.63 MB    | 2.30 MB                   |
| media    | 2.83 MB  | 2.43 MB   | 2.20 MB                  | 2.92 MB    | 2.58 MB                   |
| weather  | 6.26 MB  | 5.78 MB   | 4.16 MB                  | 6.43 MB    | 4.59 MB                   |

## Ubuntu 24.04 (WSL)

| binary   | No LTO   | LTO (fat) | LTO (fat) 1 codegen unit | LTO (thin) | LTO (thin) 1 codegen unit |
|----------|----------|-----------|--------------------------|------------|---------------------------|
| omni-led | 15.55 MB | 12.23 MB  | 10.42 MB                 | 15.62 MB   | 13.17 MB                  |
| audio    | 3.67 MB  | 2.45 MB   | 2.21 MB                  | 3.68 MB    | 2.97 MB                   |
| clock    | 3.65 MB  | 2.41 MB   | 2.19 MB                  | 3.65 MB    | 2.94 MB                   |
| media    | 4.16 MB  | 2.87 MB   | 2.65 MB                  | 4.16 MB    | 3.42 MB                   |
| weather  | 8.06 MB  | 6.28 MB   | 4.64 MB                  | 8.12 MB    | 5.56 MB                   |
